src/three/PNTSLoader.js

```diff
@@ -43,7 +43,31 @@ export class PNTSLoader extends PNTSLoaderBase {
 				} );
 
 				const geometry = new BufferGeometry();
-				geometry.setAttribute( 'position', new BufferAttribute( POSITION, 3, false ) );
+				const averageVector = new Vector3();
+				if ( POSITION ) {
+
+					geometry.setAttribute( 'position', new BufferAttribute( POSITION, 3, false ) );
+
+				} else if ( POSITION_QUANTIZED ) {
+
+					const decodedPositions = new Float32Array( POINTS_LENGTH * 3 );
+					for ( let i = 0; i < POINTS_LENGTH; i ++ ) {
+
+						for ( let j = 0; j < 3; j ++ ) {
+
+							const index = 3 * i + j;
+							decodedPositions[ index ] = ( POSITION_QUANTIZED[ index ] / 65535.0 ) * QUANTIZED_VOLUME_SCALE[ j ];
+
+						}
+
+					}
+					averageVector.x += QUANTIZED_VOLUME_OFFSET[ 0 ];
+					averageVector.y += QUANTIZED_VOLUME_OFFSET[ 1 ];
+					averageVector.z += QUANTIZED_VOLUME_OFFSET[ 2 ];
```

Comment:
nit: not a huge deal but since these are just set once we can just assign the values to the x, y, z fields.